### PR TITLE
Add Pact tests and job running them.

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?:
Add pact tests, run the verification against contracts that are already existing in the Pact broker.
Make this as a separate job and make this job as optional, so for now it will not block any PR merge.

### Which issue(s)/story(ies) does this PR fixes:
https://issues.redhat.com/browse/HAC-2988

### How to test changes / Special notes to the reviewer:
To run the pact tests, execute `make pact`.

### CODE REVIEW

**Comment by johnmcollier** [comment]
Hi @Katka92

I was looking over this PR and also https://github.com/redhat-appstudio/application-service/pull/336, and I think this PR belongs in [redhat-appstudio/application-api](https://github.com/redhat-appstudio/application-api), rather than HAS? As I’m not seeing anything HAS-specific in the tests?

File: controllers/application_pact_test_state_handlers.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+//
```
Comment (johnmcollier) [inline controllers/application_pact_test_state_handlers.go:1]: Can you add the following files to the ignore list in `codecov.yml`:
- `application_pact_test_state_handlers.go`
- `application_pact_test_utils.go`
- `start_test_env.go`

I think codecov only excludes test files ending in _test by default, so I think the above three test files will also need to be ignored.

File: controllers/start_test_env.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+)
+
+var (
+	k8sClient client.Client // You'll be using this client in your tests.
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func setupTestEnv() {
```
Comment (johnmcollier) [inline controllers/start_test_env.go:52]: What's this file needed for compared to just keeping things in `controllers/suite_test.go`?

File: controllers/application_pact_test_state_handlers.go (hunk 1 above)
Comment (Katka92) [reply controllers/application_pact_test_state_handlers.go:1]: Thank you, I've excluded those.

**Review by johnmcollier — APPROVED** [review]
Looks good 👍

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**